examples/with-reasonml/pages/_app.js

```diff
@@ -0,0 +1,28 @@
+import App, { Container } from 'next/app'
+import React from 'react'
+import { make as CountProvider } from '../components/CountProvider.bs'
+
+class MyApp extends App {
+  static async getInitialProps({ Component, router, ctx }) {
+    let pageProps = {}
+
+    if (Component.getInitialProps) {
+      pageProps = await Component.getInitialProps(ctx)
+    }
+
+    return { pageProps }
+  }
+
+  render() {
+    const { Component, pageProps, store } = this.props
+    return (
+      <Container>
+        <CountProvider>
```

Comment:
I feel like this wrapping is exactly what the Hooks API is trying to avoid.  Is there another way we can inject the provider?

Reply:
Also, we're injecting context in a place where it takes quite a leap to find.

Reply:
I don't think there is another way to inject the provider if we want it to work on every page without having to prepend it to the top level of `index.re` and `about.re`. I see this as a constraint of next being a JS framework and so have to work with its nuance.

As far as wrapping and how it relates to hooks, you still need a Provider, so I don't think there is a way to get around wrapping in terms of the context API. Maybe in the new version of reductive this will be solved.

Reply:
It seems like a lot of ceremony to avoid the use of a `ref`.   I think the `ref` can be avoided if your willing to simply replace the state on each update.  

Also, now we are introducing many more concepts, steps and indirections. 

I get that you may have to use this file eventually, but is it necessary in this case?  Should it be something that should be avoided unless really necessary?  To me, it’s akin to using `bs.raw`

Reply:
There is definitely more boilerplate here and I don't really like that at all. At the same time, you can make the same comments about `redux`. But underneath all the boilerplate are some good principles. Not that what we have here is great but just like with `redux` you learn the patterns and you move on. So I kinda see it from that perspective.